? extensions/lirc/.deps
? extensions/lirc/Makefile
? extensions/lirc/Makefile.in
? gui/Info.plist
? libltdl/COPYING.LIB
? libltdl/Makefile.am
? libltdl/Makefile.in
? libltdl/README
? libltdl/acinclude.m4
? libltdl/aclocal.m4
? libltdl/config-h.in
? libltdl/config.guess
? libltdl/config.sub
? libltdl/configure
? libltdl/configure.ac
? libltdl/configure.in
? libltdl/install-sh
? libltdl/ltdl.c
? libltdl/ltdl.h
? libltdl/ltmain.sh
? libltdl/missing
? libltdl/stamp-h.in
? server/asobj/GradientGlowFilter_as.loT
Index: plugin/klash/klash_part.cpp
===================================================================
RCS file: /sources/gnash/gnash/plugin/klash/klash_part.cpp,v
retrieving revision 1.19
diff -u -r1.19 klash_part.cpp
--- plugin/klash/klash_part.cpp	24 Aug 2007 15:49:54 -0000	1.19
+++ plugin/klash/klash_part.cpp	14 Sep 2007 22:58:31 -0000
@@ -134,7 +134,8 @@
                 m_autostart = value.toInt (&ok);
                 if (!ok)
                     m_autostart = (value.lower () == "false");
-	    }
+            }
+            m_args.push_back(name + QChar('=') + value);
         }
     }
     KParts::Part::setWidget (new KlashView (wparent));
@@ -173,21 +174,29 @@
     m_process = new KProcess;
     m_process->setUseShell (true);
     m_process->setEnvironment (QString::fromLatin1 ("SESSION_MANAGER"), QString::fromLatin1 (""));
-    QString dim;
+
+    QString cmd = procname + QString(" -x ") +
+                             QString::number(static_cast<KlashView*>(widget())->embedId());
+
     if (m_width > 0 && m_height > 0)
-        dim = QString ("-j ") + QString::number (m_width) +
-            QString (" -k ") + QString::number (m_height);
+        cmd += QString(" -j ") + QString::number(m_width) +
+               QString(" -k ") + QString::number(m_height);
+
     QString url = m_url.url();
-    QString url_param;
     if (!url.isEmpty())
-        url_param = QString ("-u ") + KProcess::quote (url);
-    QString cmd = procname + QString (" -x ") +
-        QString::number (static_cast <KlashView *> (widget ())->embedId()) +
-        QChar (' ') + dim +
-        QChar (' ') + url_param +
-        QChar (' ') + KProcess::quote (m_src_url);
+        cmd += QString(" -u ") + KProcess::quote(url);
+    url = m_docbase.url();
+    if (!url.isEmpty())
+        cmd += QString(" -U ") + KProcess::quote(url);
+
+    for (QStringList::const_iterator it=m_args.begin(), end=m_args.end();it != end; ++it)
+        cmd += QString(" -P ") + KProcess::quote(*it);
+
+    cmd += QString (" ") + KProcess::quote(m_src_url);
+
     kdDebug () << cmd << endl;
     *m_process << cmd;
+
     connect (m_process, SIGNAL (processExited (KProcess *)),
             this, SLOT (processStopped (KProcess *)));
     m_process->start (KProcess::NotifyOnExit, KProcess::NoCommunication);
Index: plugin/klash/klash_part.h
===================================================================
RCS file: /sources/gnash/gnash/plugin/klash/klash_part.h,v
retrieving revision 1.7
diff -u -r1.7 klash_part.h
--- plugin/klash/klash_part.h	1 Jul 2007 10:54:15 -0000	1.7
+++ plugin/klash/klash_part.h	14 Sep 2007 22:58:31 -0000
@@ -147,6 +147,7 @@
 protected:
     virtual bool openFile();
 private:
+    QStringList m_args;
     KlashBrowserExtension * m_browserextension;
     KlashLiveConnectExtension * m_liveconnectextension;
     KProcess * m_process;
